### PR TITLE
chore(web): lm-worker "test stub" code elimination 🎡

### PR DESCRIPTION
**Pull request by jahorton**
From https://github.com/keymanapp/keyman/pull/6533#issuecomment-1132505707:

> > After reviewing the full chain... yeah, I see why it's having to be linked the way it is now. TypeScript throws a fit if you try to directly include it because of all the Worker-space global references that aren't valid in standard document-based web code.
> 
> Many crying and tears here but basically ... yeah.

Knowing that `@ts-ignore` for single lines was a thing, I did some web searching.  Then, after seeing https://stackoverflow.com/a/59046041... I had one final idea.

> As a workaround you can use a // @ts-nocheck comment at the top of a file to disable type-checking for that file: https://devblogs.microsoft.com/typescript/announcing-typescript-3-7-beta/

I didn't get it right initially, but after much experimentation, I found a way.  By relabeling the `lm-worker` output as a `.ts` file (and yes, it **must** be a .ts file), and attaching that `@ts-nocheck` annotation at the top of the file, we can have tsc bundle the code in directly with no further complaints.

Shockingly... that doesn't have to be in a top-level build product, either - the lm-worker's can be happily included in builds via `common/predictive-text` again!

Just in case, the two commits are pretty cleanly split:

1.  I swapped out the TEMP STUB with this bundling approach in the same projects that'd been doing the TEMP STUB bit.
2. I dropped the bundling in the 'leaf' / 'top-level' projects and put it back into its original bundling location before the 🎡 chain started.

So, just in case part number 2 feels a bit risky, we can easily drop just that part and rely on the first commit.

@keymanapp-test-bot skip

### CODE REVIEW

File: common/web/input-processor/src/tsconfig.bundled.json [hunk 1]
```diff
@@ -5,7 +5,8 @@
   // for certain down-line builds if the two tsconfigs are not differentiated.
   "extends": "./tsconfig.json",
   "compilerOptions": {
-    "outFile": "../build/index.bundled.js"
+    "outFile": "../build/index.bundled.js" //,
+    //"declaration": false,
```
Comment (mcdurdin) [inline common/web/input-processor/src/tsconfig.bundled.json:9]:
```suggestion
    "outFile": "../build/index.bundled.js",
```

Note that the trailing comma is permitted in JSON-with-comments, so I left it in as it makes future changesets cleaner

File: common/web/lm-worker/build.sh [hunk 2]
```diff
@@ -35,6 +35,7 @@ display_usage ( ) {
 
 WORKER_OUTPUT=build
 WORKER_OUTPUT_FILENAME=$WORKER_OUTPUT/index.js
+WORKER_TEST_BUNDLE_TARGET_FILENAME=$WORKER_OUTPUT/index.ts-nocheck.ts
```
Comment (mcdurdin) [inline common/web/lm-worker/build.sh:38]: I would like a clearer name than `index.ts-nocheck.ts`. That doesn't describe the what module is but instead just the hack that it uses to make it work :grin:

**Review by mcdurdin — APPROVED** [review]
I think this is all good as well. I am a little bit uncomfortable with referencing a file in a build artifact from another project within tsconfig.json, but it's less hacky than the previous round.

File: common/web/lm-worker/build.sh (hunk 2 above)
Comment (jahorton) [reply common/web/lm-worker/build.sh:38]: I would too, honestly.  Just, I was drawing blanks as to what that name would even be.

Comment (mcdurdin) [reply common/web/lm-worker/build.sh:38]: How about `index.wrapped-for-bundle.ts` (long but points in the right direction, and as the name is internal to the build process, it won't leak into distribution)

**Comment by jahorton** [comment]
This morning, I asked in the TypeScript Discord if there were any cleaner way... but alas, it's pretty much either this, our previous approach, or committing a dummy wrapper function with a unique, identifying string inside of it.

With the dummy wrapper function, we'd do a post-build search-and-replace to insert the worker code completely outside of the standard `tsc` compilation process.  Rather than just `cat`-ing the code files together.

**Comment by jahorton** [comment]
> tsc doesn't support configurable (or any, really) calling of pre-build or post-build scripts.

... which has led me to make a formal feature request of TS:  https://github.com/microsoft/TypeScript/issues/49225

I doubt we'll see anything soon, but I figured we may as well raise the issue.

**Comment by mcdurdin** [comment]
Failed iOS samples build looks like it may be due to keyman-version not being built in web/source/build.sh?

**Comment by jahorton** [comment]
> Failed iOS samples build looks like it may be partly due to keyman-version not being built in web/source/build.sh?

Ah, that'd have been addressed later on one of the base branches; that was a commit on #6625 after the rebases + force-pushes I did.

**Pull request merged**